pkg/engine/mutate/patch/buffer.go

```diff
@@ -0,0 +1,20 @@
+package patch
+
+import "bytes"
+
+type buffer struct {
+	*bytes.Buffer
+}
+
+func (buff buffer) UnmarshalJSON(b []byte) error {
+	buff.Reset()
+	_, err := buff.Write(b)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (buff buffer) MarshalJSON() ([]byte, error) {
+	return buff.Bytes(), nil
+}
```

Comment:
This design has a subtle expectation: that callers will copy or discard the `[]byte` result of `buffer.MarshalJSON` before calling `buffer.UnmarshalJSON` again on the same `buffer`.

Reply:
added comments to caution usage